main/zoe/guide/contracts/autoswap.md

```diff
@@ -2,161 +2,170 @@
-:::
+Autoswap is a contract that maintains a pool of assets (the 'liquidity pool') that can
+always trade against any incoming offer. It can do this because it uses a rule to set
+the prices, so the price is updated as its assets change.  This contract uses the
+[constant-product](https://github.com/runtimeverification/verified-smart-contracts/blob/uniswap/uniswap/x-y-k.pdf)
+rule.
+
+This contract follows the design of [UniSwap](https://uniswap.org/), with a single
+pool. This means any installation of autoswap can make exchanges between two
+issuers. Our multipoolAutoswap generalizes this to many pools, all of which share a
+common intermediate pool. We leave this single autoswap contract available because it's
+simpler and therefore easier to read. We expect all practical usage to migrate to
+multipoolAutoswap.
+
+## The Autoswap API
+
+When the contract is instantiated, the two tokens (`Central` and `Secondary`) are
+specified in the `issuerKeywordRecord`. There is no behavioral difference between the
+two when trading; the names were chosen for consistency with multipoolAutoswap. When
+trading, use the keywords `In` and `Out` to specify the amount to be paid in and the
+amount to be received.
+
+When adding or removing liquidity, the amounts deposited must be in proportion to the
+current balances in the pool. The amount of the `Central` asset is used as the
+basis. The `Secondary` assets must be added in proportion.  If less `Secondary` is
+provided than required, we refuse the offer. If more is provided than is required, we
+return the excess.
+
+Before trading can take place, someone must add liquidity using
+`makeAddLiquidityInvitation()`. Separate invitations are available for adding and
+removing liquidity, and for doing swaps. Other API operations support price checks and
+checking the size of the liquidity pool.
+
+The `swap()` operation requires either the input amount or the output amount to be
+specified. `makeSwapInInvitation()` treats the give amount as definitive, while
+`makeSwapOutInvitation()` honors the want amount. With swapIn, a want amount can be
+specified, and if the offer can't be satisfied, the offer will be refunded. Similarly
+with swapOut, the want amount will be satisfied if possible. If more is provided as the
+give amount than necessary, the excess will be refunded. If not enough is provided, the
+offer will be refunded.
+
+The `publicFacet` can make new invitations (`makeSwapInInvitation()`,
+`makeSwapOutInvitation()`, `makeAddLiquidityInvitation()`, and
+`makeRemoveLiquidityInvitation()`), tell how much would be paid for a given input
+(`getInputPrice()`), or how much is required to be deposited in order to get a
+specified amount out (`getOutputPrice()`). In addition, there are requests for the
+Liquidity issuer (`getLiquidityIssuer()`), the current outstanding liquidity
+(`getLiquiditySupply()`), and the current balances in the pool (`getPoolAllocation()`).
```

Comment:
Well, that got a bit long. I'd suggest breaking this up into a list, with entries for
new invitations, how much paid for input, how much deposited, and those three final requests.

Reply:
done.